corehq/motech/openmrs/finders.py

```diff
@@ -155,12 +177,16 @@ def find_patients(self, requests, case, case_config):
             return []
         patients_scores = sorted(candidates.values(), key=lambda cand: cand.score, reverse=True)
         if len(patients_scores) == 1:
-            return [patients_scores[0].patient]
+            patient = patients_scores[0].patient
+            self.save_match_id(case, case_config, patient)
```

Comment:
`self.save_match_id` will save the patient's value of the identifier(s) listed in the `OpenmrsCaseConfig`'s "id_matchers" to the case, so that next time the case's patient will be matched immediately. An example value of OpenmrsCaseConfig.id_matchers would look like:
```json

  "id_matchers": [
    {
      "case_property": "openmrs_id", 
      "identifier_type_id": "8d793bee-c2cc-11de-8d13-0010c6dffd0f", 
      "doc_type": "IdMatcher"
    }
  ], 
```